### PR TITLE
AIP-51 - Local Executor Compatibility

### LINKED ISSUE #27928: AIP-51 - Local Executor Compatibility
### Overview
Executors that run locally have certain limitations and requirements, many of which are currently hardcoded in core Airflow code. To add a new Executor, that would run locally, one would be required to change this core Airflow code.
### Examples
- **1a**) Whether or not to start the serve_logs sub-process, currently hardcoded to LocalExecutor and SequentialExecutor: https://github.com/apache/airflow/blob/27e2101f6ee5567b2843cbccf1dca0b0e7c96186/airflow/cli/commands/scheduler_command.py#L93
- **1b**) When running in standalone mode, asserts a local Executor is being used, currently hardcoded to LocalExecutor and SequentialExecutor: https://github.com/apache/airflow/blob/26f94c5370587f73ebd935cecf208c6a36bdf9b6/airflow/cli/commands/standalone_command.py#L156-L158
- **1c**) Local Executors need a copy of Airflow configuration created in tmp, currently hardcoded to LocalExecutor and SequentialExecutor: https://github.com/apache/airflow/blob/26f94c5370587f73ebd935cecf208c6a36bdf9b6/airflow/jobs/backfill_job.py#L505-L509
### Proposal
A static method or attribute on the Executor class which can be checked by core code.

There is a precedent already set with the `supports_ad_hoc_ti_run` attribute, see:
https://github.com/apache/airflow/blob/fb741fd87254e235f99d7d67e558dafad601f253/airflow/executors/kubernetes_executor.py#L435 https://github.com/apache/airflow/blob/26f94c5370587f73ebd935cecf208c6a36bdf9b6/airflow/www/views.py#L1735-L1737

### PR DESCRIPTION
**Pull request by pierrejeambrun**
closes: https://github.com/apache/airflow/issues/27928

### CODE REVIEW

File: airflow/executors/base_executor.py [hunk 1]
```diff
@@ -69,6 +69,8 @@ class BaseExecutor(LoggingMixin):
     job_id: None | int | str = None
     callback_sink: BaseCallbackSink | None = None
 
+    is_local: bool = False
```
Comment (o-nikolas) [inline airflow/executors/base_executor.py:72]: We should add a simple unit test that asserts that this field is present with the default that we expect (and likewise add one for the specific executors which override it to ensure we get the expected value there). If someone were to accidentally (or misguidedly change it with intention) modify one of these we'd like tests to fail.

**Review by o-nikolas — CHANGES_REQUESTED** [review]
Thanks so much for jumping on board with AIP-51 Pierre! :smile:

File: airflow/executors/base_executor.py (hunk 1 above)
Comment (pierrejeambrun) [reply airflow/executors/base_executor.py:72]: Good idea, i'll do that :)

Comment (pierrejeambrun) [reply airflow/executors/base_executor.py:72]: Done :)

**Comment by o-nikolas** [comment]
Hey @pierrejeambrun!

#27974 was merged into main, you can now make use of it in this PR :)

**Comment by pierrejeambrun** [comment]
@o-nikolas Thanks for the heads up. I updated the PR on sunday and also added the test you mentionned. Did I miss something on the update part ?

**Comment by o-nikolas** [comment]
> @o-nikolas Thanks for the heads up. I updated the PR on sunday (mostly to use `import_default_executor_cls`) and also added the test you mentionned. Let me know if I missed something on update part :)

Nope, I was just too eager :sweat_smile: I'll have a review today! :pray:

**Review by o-nikolas — APPROVED** [review]
An absolutely beautiful PR with the recent changes :rocket:

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
VERY cool :)

**Pull request merged**